### PR TITLE
Tracking accounts/tokens changes in user settings should reflect on the dashboard

### LINKED ISSUE #325: After adding assets from the user settings pane they are not reflected in the dashboard
## Problem Definition

After adding assets from the user settings pane, like an ETH account, a BTC account or ETH tokens, if the user navigates back to the dashboard the new assets are not reflected in the dashboards "All assets" table.

## Task

- [ ] Fix this problem. If assets are added at the user settings they should be reflected to the dashboard when the user goes back there.
- [ ] Add tests for this.

### PR DESCRIPTION
**Pull request by LefterisJP**
- Adding a BTC/ETH account should have balances reflected in dashboard
- Removing a BTC/ETH account should have balances reflected in dashboard
- Tracking a new token should have balances reflected in dashboard
- Stoping to track a new token should have balances reflected in dashboard

@kelsos could you have a look? Also ... would it be too much to ask for writting tests for these? Or you think they will need to be deleted once the vue rewrite happens?

Fix #325

### CODE REVIEW

File: ui/balances_table.ts [hunk 1]
```diff
@@ -43,18 +43,36 @@ function create_full_data(): AssetBalance[] {
         if (!inter_data.hasOwnProperty(asset)) {
             continue;
         }
-        total_usd += inter_data[asset]['usd_value'] as number;
+        let asset_usd_value;
+        if (typeof inter_data[asset].usd_value === 'string') {
```
Comment (LefterisJP) [inline ui/balances_table.ts]: @kelsos is there any other way in typescript to do this? Feels not so clean.

File: ui/balances_table.ts [hunk 2]
```diff
@@ -43,18 +43,36 @@ function create_full_data(): AssetBalance[] {
...
+
     const full_data: AssetBalance[] = [];
     // we should have all balances by now
     for (const asset in inter_data) {
         if (!inter_data.hasOwnProperty(asset)) {
             continue;
         }
-        const amount = inter_data[asset].amount as number;
+        let amount;
+        if (typeof inter_data[asset].amount  === 'string') {
```
Comment (LefterisJP) [inline ui/balances_table.ts]: same here and below.

**Comment by kelsos** [comment]
I will have a look either at a break or after the conference today. But in general we can get the test logic there and then repurpose the and adjust them to the new setup

File: ui/balances_table.ts (hunk 1 above)
Comment (kelsos) [reply ui/balances_table.ts]: I tried the classic JavaScript approach of multiplying by 1 but it doesn't seem to work with the TypeScript type system.

Comment (kelsos) [reply ui/balances_table.ts]: For the Vue rewrite the parsing is happening on the edges so there will not be such a problem

**Comment by LefterisJP** [comment]
What about tests for this?

File: ui/balances_table.ts (hunk 1 above)
Comment (LefterisJP) [reply ui/balances_table.ts]: that's great. I actually can't wait :)

**Comment by LefterisJP** [comment]
@kelsos if you can write some tests for this that would be great. Essentially tests that make sure the data are updated after adding/removing stuff in user settings both in that same page but also in the dashboard when navigating to/back from it.

In the meantime I will merge this since there is one similar issue I want to solve: https://github.com/rotkehlchenio/rotkehlchen/issues/392

**Comment by kelsos** [comment]
@LefterisJP Sure can you open an issue about the tests and assign it to me?

**Comment by LefterisJP** [comment]
@kelsos done! https://github.com/rotkehlchenio/rotkehlchen/issues/396

**Pull request merged**